crypto/data_formats.go

```diff
@@ -15,41 +15,72 @@
 package crypto
 
 import (
-	"encoding/base64"
 	"fmt"
-	"strconv"
 
-	"github.com/benlaurie/objecthash/go/objecthash"
-	"github.com/google/trillian"
+	"github.com/google/certificate-transparency-go/tls"
+
+	pb "github.com/google/trillian"
 )
 
 // This file contains struct specific mappings and data structures.
-// TODO(gdbelvin): remove data-structure specific operations.
 
-// Constants used as map keys when building input for ObjectHash. They must not be changed
-// as this will change the output of hashRoot()
+type logVersion int
+
+const (
+	// LogRootV0 indicates version 0 for the canonical representation of the SignedLogRoot.
+	LogRootV0 logVersion = iota
+)
+
+// CanonicalLogRoot returns a canonical TLS serialization of the log root.
```

Comment:
`SerializeLogRoot` might better describe what the function does (but fine as-is if you prefer)

Reply:
done